File: frontend/src/manager/matchPlacementLabel.ts
/** Libellé brush des places en jeu (ex. 1-2, 5-6) depuis le champ tour. */
export function matchPlacementLabel(tour: string): string | null {
  const normalized = tour.trim().toLowerCase();

  if (normalized === "finale") return "1-2";
  if (normalized === "petite finale") return "3-4";

  const match = tour.match(/classement\s+(\d+)\s*-\s*(\d+)/i);
  if (!match) return null;

  const winnerPlace = Number.parseInt(match[1], 10);
  const loserPlace = Number.parseInt(match[2], 10);

  if (loserPlace - winnerPlace !== 1) return null;

  return `${winnerPlace}-${loserPlace}`;
}

export function parsePlacementTour(
  tour: string
): { winnerPlace: number; loserPlace: number } | null {
  const label = matchPlacementLabel(tour);
  if (!label) return null;

  const [winnerPlace, loserPlace] = label.split("-").map((value) =>
    Number.parseInt(value, 10)
  );

  return { winnerPlace, loserPlace };
}

File: engine/match_placement_label.py
"""Libellés de places en jeu (ex. 1-2, 5-6) depuis le champ tour."""

from __future__ import annotations

import re

_CLASSEMENT_RE = re.compile(r"classement\s+(\d+)\s*-\s*(\d+)", re.IGNORECASE)


def match_placement_label(tour: str) -> str | None:
    normalized = tour.strip().lower()

    if normalized == "finale":
        return "1-2"
    if normalized == "petite finale":
        return "3-4"

    match = _CLASSEMENT_RE.search(tour)
    if not match:
        return None

    winner_place = int(match.group(1))
    loser_place = int(match.group(2))

    if loser_place - winner_place != 1:
        return None

    return f"{winner_place}-{loser_place}"


def parse_placement_tour(
    tour: str,
) -> dict[str, int] | None:
    label = match_placement_label(tour)
    if not label:
        return None

    winner_place, loser_place = (int(value) for value in label.split("-"))
    return {"winnerPlace": winner_place, "loserPlace": loser_place}
